### PR TITLE
#166980212 Rating articles

### PR DESCRIPTION
**Pull request by jmusila**
### What does this PR do?
Adds Rating Articles Functionality
### Description of Task to be completed?
   - [x] add five-star rating 
### How should this be manually tested?
- Checkout to:

      ft-rate-article-166980212
     
- Install dependencies:

      yarn
     
- Run local server:

      yarn develop

- Click on rating stars at the footer of the article
- Confirm your rating by making sure that the rate average on the artcile header has increased or decreased
### Any background context you want to provide?
None
### What are the relevant pivotal tracker stories?
#166980212
### Screenshots (if appropriate)
Screenshot 2019-07-19 at 07 17 42

### CODE REVIEW

**Review by wangonya — COMMENTED** [review]
Tested locally. The rating functionality works.

File: src/containers/articles/Article.js [hunk 1]
```diff
@@ -14,29 +15,39 @@ class ArticlePage extends Component {
   }
 
   handleEditClick = () => {
-    const { articleData, getArticle, history } = this.props; // eslint-disable-line
+  const { articleData, getArticle, history } = this.props; // eslint-disable-line
     getArticle(articleData.slug);
     history.push('/articles/update');
```
Comment (FlevianK) [inline src/containers/articles/Article.js]: This action should occur when this `getArticle(articleData.slug);` is successful, your implementation has not catered that?

Comment (ceexon) [reply src/containers/articles/Article.js]: The article page cannot be loaded unless article data has been found 
If the article data is not there then that page will remain to be in a loading state and thus neither the article nor the buttons would show up.

**Comment by ceexon** [comment]
@FlevianK I mistakenly resolved the conversation there was on using history in async

File: src/actions/__tests__/passwordconfirm.test.js [hunk 2]
```diff
@@ -26,4 +30,12 @@ describe('Reset confirm actions', () => {
     };
     expect(resetConfirmError(data)).toEqual(expectedAction);
   });
+  test('should fail', () => {
+    instance.post.mockImplementation(() => Promise.reject({}));
```
[the remark this answers is not in the document]
Comment (FlevianK) [reply src/actions/__tests__/passwordconfirm.test.js:34]: Can we fix this once and for all:

e.g ... => Promise.reject(new TypeError("something bad happened"))

File: src/actions/__tests__/passwordreset.test.js [hunk 3]
```diff
@@ -1,5 +1,9 @@
-import { resetPasswordRequest, resetPassword, resetPasswordError } from '../passwordResetActions';
+import {
```
[the remark this answers is not in the document]
Comment (FlevianK) [reply src/actions/__tests__/passwordreset.test.js:1]: Fix this

**Review by FlevianK — APPROVED** [review]
Make those small changes, otherwise LGTM

**Comment by jmusila** [comment]
> Make those small changes, otherwise LGTM

@FlevianK I already resolved the requested reviews, I think everything is okay apart from the hound comments.

**Review by ceexon — APPROVED** [review]

File: src/Main.scss [hunk 4]
```diff
@@ -69,3 +69,11 @@ button.article-hover-title {
   font-size: 18px;
   font-weight: normal;
 }
+.article-hover-title:hover {
```
Comment (FlevianK) [inline src/Main.scss]: Remove this rule, it is a repetition

File: src/reducers/ratingArticles.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import actionTypes from '../actions/types';
+
+const initialState = {
+  data: {},
+  error: null,
+  loading: false,
+};
+
+export default function (state = initialState, action) {
```
Comment (FlevianK) [inline src/reducers/ratingArticles.js]: Use es6 :

e.g export default (state = initialState, action) => { ...

Comment (jmusila) [reply src/reducers/ratingArticles.js]: Done

File: src/Main.scss (hunk 4 above)
Comment (jmusila) [reply src/Main.scss]: Done

**Pull request merged**